app/presenters/tree_builder_report_reports.rb

```diff
@@ -26,43 +27,38 @@ def root_options
     }
   end
 
-  # Get root nodes count/array for explorer tree
+  # Get root node and all children report folders. (will call get_tree_custom_kids to get report details)
   def x_get_tree_roots(count_only, options)
-    objects = []
-    @rpt_menu.each_with_index do |r, i|
-      objects.push(
-        :id    => i.to_s,
-        :text  => r[0],
-        :icon  => "pficon #{@grp_title == r[0] ? 'pficon-folder-close-blue' : 'pficon-folder-close'}",
-        :tip   => r[0]
-      )
+    @rpt_menu.each_with_index.each_with_object({}) do |(r, node_id), a|
```

Comment:
I thought `x_get_tree_roots` only called with `count_only = true` in rare instances where trees are build from trees.

I added a fix.
Thanks